bin/add_transform_test.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+"""Automatically create transform tests input and output files given an input template.
+
+Usage:
+    add_transform_test.py --template-file=sam-template.yaml
+
+Options:
+    --template-file=<i>       Location of SAM template to transform [default: template.yaml].
+
+"""
+import re
+import os
+import shutil
+import yaml
+from docopt import docopt
+from pathlib import Path
+
+from samtranslator.yaml_helper import yaml_parse
+from sam_translate import transform_template
+
+cwd = os.getcwd()
+CLI_OPTIONS = docopt(__doc__)
+
+
+def get_input_file_path():
+    input_file_option = CLI_OPTIONS.get("--template-file")
+    return os.path.join(cwd, input_file_option)
+
+
+def copy_input_file_to_transform_test_dir(input_file_path, transform_test_input_path):
+    shutil.copyfile(input_file_path, transform_test_input_path)
+
+    replace_arn_partitions(transform_test_input_path)
```

Comment:
One thing I often encountered was with `AWS::ApiGateway::RestApi` needing (see any of the API transform tests e.g. `api_with_apikey_required`)

```json
        "EndpointConfiguration": {
          "Types": [
            "REGIONAL"
          ] 
        },
        "Parameters": {
          "endpointConfigurationTypes": "REGIONAL"
        }
```

Maybe we could add that by default?